### PR TITLE
Improve search

### PR DESCRIPTION
**Pull request by shady2k**
Added parent block and grouping by it.
image

### CODE REVIEW

**Comment by shady2k** [comment]
@WuTheFWasThat, can you look at this?

**Comment by WuTheFWasThat** [comment]
hmm, I'm not really convinced this is a good idea, since the space could be used by more results, and as a document gets a bit bigger most results will be under separate parents.  

BTW, search becomes unusable once your document gets long :( and I generally recommend searching by marks instead.  Would love to have faster search but it's a bit involved to implement

**Comment by shady2k** [comment]
But the conclusion without groups is absolutely inconvenient. Especially if you come across the same text in different lines. Maybe make this behavior optional in the settings?

**Comment by WuTheFWasThat** [comment]
ok sounds good, i'd prefer default off if that's ok.  you could add it to either ClientSettings or LocalDocSettings or DocSettings, not sure which one is best

then maybe it works to just change one line to `path.parent && setting_is_on ? await session.document.getLine(path.parent.row) : []`

File: src/assets/ts/components/settings.tsx [hunk 1]
```diff
@@ -117,11 +117,13 @@ export default class SettingsComponent extends React.Component<Props, State> {
         return [
           {
             contents: 'Some blah result'.split(''),
+            contentsParent: 'Some blah result'.split(''),
```
Comment (WuTheFWasThat) [inline src/assets/ts/components/settings.tsx:120]: maybe rename everywhere to parentContents

**Pull request closed without merging**